apps/prairielearn/src/ee/lib/aiQuestionGeneration.ts

```diff
@@ -216,49 +226,101 @@ Keep in mind you are not just generating an example; you are generating an actua
       user: openAiUserFromAuthn(authnUserId),
     });
 
-    extractFromCompletion(completion, job);
-    const html = job?.data?.html;
-
-    job.data['initialGenerationErrors'] = [];
-    job.data['prompt'] = userPrompt;
-    job.data['generation'] = completion.choices[0].message.content;
-    job.data['context'] = context;
-    job.data['completion'] = completion;
+    const results = extractFromCompletion(completion, job);
+    const html = results?.html;
 
+    let errors: string[] = [];
     if (html && typeof html === 'string') {
-      const errors = validateHTML(html, false, !!job?.data?.python);
-      job.data['initialGenerationErrors'] = errors;
-      job.data['finalGenerationErrors'] = errors;
-      if (errors.length > 0) {
-        await regenInternal(
-          job,
-          client,
-          authnUserId,
-          userPrompt,
-          `Please fix the following issues: \n${errors.join('\n')}`,
-          html,
-          typeof job?.data?.python === 'string' ? job?.data?.python : undefined,
-          0,
-          false,
-        );
-      }
+      errors = validateHTML(html, false, !!results?.python);
     } else {
+      errors = ['Please generate a question.html file.'];
+    }
+
+    let questionQid = '';
+
+    if (saveLocals) {
```

Comment:
What does "locals" mean in this context? That pretty much always refers to `res.locals`, but that doesn't appear to be the case here.

Edit: maybe it does, but this is still confusing. I interpreted the name as a boolean like "should the locals be saved", but that's not what's happening. Maybe the "save" prefix is tripping me up?

Reply:
agreed- i used the locals for questioneditor calls, but due to the switch to the rpc routes, i switched to just including extra info in a94d9c043b56e640362369ff73f52d0a019803dd.